framework/helpers/BaseArrayHelper.php

```diff
@@ -247,6 +247,33 @@ public static function remove(&$array, $key, $default = null)
     }
 
     /**
+     * Search and removes an item from an array and returns the key if found. 
+     * If the value does not exists it will return the default value.
+     *
+     * Example,
+     *
+     * ```php
+     * $array = ['value1', 'value2'];
+     * $key = \yii\helpers\ArrayHelper::removeValue($array, 'value2');
+     * // result:
+     * // $array = ['value1']
+     * ```
+     *
+     * @param $array
+     * @param $value
+     * @param null $default
+     * @return mixed|null
+     */
+    public static function removeValue(&$array, $value, $default = null)
```

Comment:
Could it not be confused by the functionality of the `array_pop` function and expect that it will only remove the last occurrence? I've no problem renaming it, but in my opinion removeValue tells you exactly what it does.